security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -443,6 +450,23 @@ func (sf *SecretFetcher) FindIngressGatewaySecret(key string) (secret model.Secr
 	val, exist := sf.secrets.Load(key)
 	secretFetcherLog.Debugf("load secret %s from secret fetcher: %v", key, exist)
 	if !exist {
+		// In some case, we might not get the secret from watch. Try to directly obtain the secret from API call
+		// as a remedy. If we can find a secret, add it to the cache
+		if sf.coreV1 != nil {
+			if secret, err := sf.coreV1.Secrets(sf.secretNamespace).Get(key, metav1.GetOptions{}); err == nil {
+				log.Infof("Find secret %s by direct api call", key)
+				sf.AddSecret(secret)
```

Comment:
That sf.AddSecret is used for test only, please update its comment accordingly.
Calling sf.AddSecret here causes an extra secret push to one connection, which would cause problem. When SDS client sends one request over the connection, we should only push one response back.

The existing caller to FindIngressGatewaySecret is SecretCache.generateSecret(), which is called by StreamSecrets that will push the secrets.
The sf.scrtAdded, once called, will also find SDS connection by cache and push the fetched secret to that SDS connection.

Reply:
Yes, I noticed that double push in my test, but it seems not causing any issue, and I don't see an existing mechanism to prevent double push. If it does matter, I can try to avoid that. I also mentioned another solution in my reply above, which is to return secret but not putting it into the cache. That would have less impact to the current flow, and will not trigger the double push (unless watcher got the secret soon after).  If sf.AddSecret() is used for test only, then that is probably a better solution. How do you think?

Reply:
Updated the patch so that it doesn't add the secret to cache.